tests/integration/targets/.gitignore

```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```

Comment:
I'm not sure why you would want to ignore everything but `.gitignore` here. How are folks supposed to add tests to the repo without jumping through some hoops?

Why not simply create `.gitkeep` files, or empty `.gitignore` files?

Reply:
Good point, adding tests would mean changing `.gitignore`. That doesn't make much sense to me.

The usual way (that I know) is to have a (empty) `.gitkeep` file if you want to enforce that a directory exists. I think this would be even better than an empty `.gitignore`.

Reply:
I guess I didn't sleep enough. You're right, it shouldn't ignore things. I'd still got for `.gitignore` because there's like a million naming conventions. Somebody uses `.gitdir` or something else. I'd rather use a file that is meaningful to Git.

Reply:
`.gitkeep` isn't meaningful to git, but that's just the point imho. It's not special for git, but it tells _people_ it's only there to keep the directory.

Reply:
personally, I am a fan of using a `.gitignore` in place of a `.gitkeep` or other file, when the `.gitignore` has or will have some other use for that directory. But I'm not certain that it definitely will in this case, so I might prefer a `.gitkeep` (whatever name)? In the end I don't have a strong opinion on it, and I don't think the performance penalty would be perceptible, so I'm fine with whatever you all choose.

I do agree it this one should not be ignoring files ;)

Perhaps whether it's a `.gitkeep` or a `.gitignore`, we should add comments inside that explain the file is not needed once actual tests are in place?

Reply:
I agree that it should be `.gitkeep` or `.keep`. This is not the correct use for a `.gitignore`. `.gitignore`s are meant for well... telling git to *ignore* files, so having a blank one just so git does *not* ignore a directory seems silly and potentially confusing.

> Perhaps whether it's a .gitkeep or a .gitignore, we should add comments inside that explain the file is not needed once actual tests are in place?

+1 to that.

Reply:
> but it tells _people_ it's only there to keep the directory.

Agreed. Such things should be targeting humans.

> The changelog test in ansible-core supports both `.keep` and `.gitkeep` in `changelogs/fragments/` (all other files are expected to be YAML files not starting with `.`), for example, but no other variants.

Interestingly, it is usually a good idea to have a `.gitignore` file in the change fragments directories, specifically to prevent people from accidentally adding unintended garbage there.

> Perhaps whether it's a `.gitkeep` or a `.gitignore`, we should add comments inside that explain the file is not needed once actual tests are in place?

Yes! That's what I was about to go for after hearing all of this feedback :)